clients/app/src/views/screens/Event/components/EventComponent.tsx

```diff
@@ -21,12 +21,41 @@ import Icon from 'views/components/base/Icon';
 import { Icons } from 'assets/icons';
 import { getReadableNumber } from 'utils/helpers/readableNumber';
 import { IEventComponentProps } from '../types';
+import buildEventMembersStore from 'stores/eventMembers';
 
 const EventComponent: React.FC<IEventComponentProps> = ({ data }) => {
   const { pushTo } = useNavigationWithParams();
   const { colors } = useTheme();
   const [ratio, setRatio] = React.useState(3 / 2);
 
+  const eventMembersStore = React.useMemo(() => buildEventMembersStore(), []);
+  const { add: storeAdd, delete: storeDelete } = eventMembersStore.useSelector(
+    'add',
+    'delete'
+  );
+
+  const [isJoined, setIsJoined] = React.useState(data.post.joined || false);
+
+  const onJoinLeavePress = React.useCallback(() => {
+    setIsJoined(!isJoined);
+
+    if (isJoined) {
+      eventMembersStore.remove({ eventId: data.id });
+    } else {
+      eventMembersStore.setFieldValue('eventId', data.id);
+      eventMembersStore.create();
+    }
+  }, [data.id, isJoined, eventMembersStore]);
+
+  React.useEffect(() => {
```

Comment:
Isn't it better to write this part like this?

```
if (storeAdd.status === 'error')
  setIsJoined(!isJoined);
```

Reply:
same in PR: https://github.com/ultras-io/ultras/pull/228